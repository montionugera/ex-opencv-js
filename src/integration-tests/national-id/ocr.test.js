import * as fp from 'lodash/fp'
import fs from 'fs'
import path from 'path'
import glob from 'glob'
import { ServiceBroker } from 'moleculer'

const filterList = ['1.jpg']

describe('ocr', () => {
  const broker = new ServiceBroker({
    logLevel: 'fetal',
    middlewares: [require('lib/services/__mocks__/tracing')],
  })

  broker.loadService(path.join(__dirname, '../../handler/process.js'))

  const files = glob.sync(
    path.join(__dirname, '../../__fixtures__/national-id/*.jpg')
  )

  const testCases = fp.pipe(
    fp.filter((path) => {
      if (fp.isEmpty(filterList)) return true

      const name = fp.pipe(
        fp.split('/'),
        fp.last
        //
      )(path)

      return fp.contains(name, filterList)
    }),
    fp.map((path) => {
      const name = fp.pipe(
        fp.split('/'),
        fp.last
        //
      )(path)
      const process = () =>
        broker.call('ocr.process', {
          type: 'national_id',
          files: [
            {
              buffer: fs.readFileSync(path),
            },
          ],
        })

      return {
        name,
        process,
      }
    })
  )(files)

  beforeAll(async () => {
    await broker.start()
  })

  afterAll(async () => {
    await broker.stop()
  })

  testCases.forEach(({ name, process }) => {
    describe(name, () => {
      it('return correctly', async () => {
        const result = await process()
        expect(result).toMatchSnapshot()
      }, 30000)
    })
  })
})
