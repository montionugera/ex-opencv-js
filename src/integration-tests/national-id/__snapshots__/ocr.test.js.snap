// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ocr 1.jpg return correctly 1`] = `
Object {
  "address_th_1": "14 หมู่ที่ 10 ต.พังเคน อ.นาตาล",
  "address_th_2": "จ.อุบลราชธานี",
  "confidence": Object {
    "address_th_1": 0.97,
    "address_th_2": 0.94,
    "date_of_birth_en": 0.94,
    "date_of_birth_th": 0.91,
    "date_of_expiry_en": 0.94,
    "date_of_expiry_th": 0.89,
    "date_of_issue_en": 0.75,
    "date_of_issue_th": 0,
    "first_name_en": 0.89,
    "id_number": 0.95,
    "last_name_en": 0.98,
    "name_th": 0.7,
    "religion_th": 1,
    "serial_number": 0.98,
  },
  "date_of_birth_en": "10 Apr. 1936",
  "date_of_birth_th": "10 เม.ย. 2479",
  "date_of_expiry_en": "LIFELONG",
  "date_of_expiry_th": "ตลอดชีพ",
  "date_of_issue_en": "20 Any 2018",
  "date_of_issue_th": "",
  "first_name_en": "Mrs. Tui",
  "id_number": "5 3430 00028 399",
  "last_name_en": "Chawmuangkhong",
  "name_th": "นาง ตุ๊ย ชาวเมืองคง",
  "religion_th": "พุทธ",
  "serial_number": "3430-01-08200931",
}
`;

exports[`ocr 2.jpg return correctly 1`] = `
Object {
  "address_th_1": "335/62 หมู่ที่ 9 ต.สุรนารี อ.เมืองนครราชสีมา",
  "address_th_2": "",
  "confidence": Object {
    "address_th_1": 0.92,
    "address_th_2": 0,
    "date_of_birth_en": 0.9,
    "date_of_birth_th": 0.98,
    "date_of_expiry_en": 0.82,
    "date_of_expiry_th": 0.92,
    "date_of_issue_en": 0.87,
    "date_of_issue_th": 0.49,
    "first_name_en": 0.93,
    "id_number": 0.9,
    "last_name_en": 0.86,
    "name_th": 0.99,
    "religion_th": 0,
    "serial_number": 0,
  },
  "date_of_birth_en": "31 Aug. 1989",
  "date_of_birth_th": "31 ส.ค. 2532",
  "date_of_expiry_en": "30 Aug. 2021",
  "date_of_expiry_th": "30 ส.ค. 2554",
  "date_of_issue_en": "11 Oct. 2012",
  "date_of_issue_th": "11 ตค. 60",
  "first_name_en": "Mr. Naphong",
  "id_number": "1 1008 00573 59 2",
  "last_name_en": "Udomphonworarak",
  "name_th": "นาย ณพงศ์ อุดมพรวรรักษ์",
  "religion_th": "",
  "serial_number": "",
}
`;

exports[`ocr 3.jpg return correctly 1`] = `
Object {
  "address_th_1": "14 หมู่ที่ 7 ต.หนองระเวียง อ.เมืองนครราชสีมา",
  "address_th_2": "จ.นครราชสีมา",
  "confidence": Object {
    "address_th_1": 0.98,
    "address_th_2": 0.91,
    "date_of_birth_en": 0.99,
    "date_of_birth_th": 0.98,
    "date_of_expiry_en": 0.87,
    "date_of_expiry_th": 0.98,
    "date_of_issue_en": 0.55,
    "date_of_issue_th": 0.9,
    "first_name_en": 0.97,
    "id_number": 0.9,
    "last_name_en": 0.85,
    "name_th": 0.96,
    "religion_th": 0,
    "serial_number": 0.95,
  },
  "date_of_birth_en": "22 Feb. 2002",
  "date_of_birth_th": "22 ก.พ. 2545",
  "date_of_expiry_en": "21 Feb. 2026",
  "date_of_expiry_th": "21 ก.พ. 2569",
  "date_of_issue_en": "7 Aug. 2017",
  "date_of_issue_th": "7 ส.ค. 2560550",
  "first_name_en": "Miss Krusumakorn",
  "id_number": "1 3099 02941 82 1099",
  "last_name_en": "Faothalsong",
  "name_th": "น.ส. กรุสุมากร เฝ้าไธสงค์",
  "religion_th": "",
  "serial_number": "3001-02-08071436",
}
`;

exports[`ocr 4.jpg return correctly 1`] = `
Object {
  "address_th_1": "71 หมู่ที่ 3 ต.หนองเสม็ด อ.เมืองตราด",
  "address_th_2": "จ.ตราด",
  "confidence": Object {
    "address_th_1": 0.95,
    "address_th_2": 0.97,
    "date_of_birth_en": 0.93,
    "date_of_birth_th": 0.99,
    "date_of_expiry_en": 0.89,
    "date_of_expiry_th": 0.92,
    "date_of_issue_en": 0.81,
    "date_of_issue_th": 0.96,
    "first_name_en": 0.99,
    "id_number": 0.85,
    "last_name_en": 0.98,
    "name_th": 0.97,
    "religion_th": 0.32,
    "serial_number": 0.88,
  },
  "date_of_birth_en": "2 Jun. 2000",
  "date_of_birth_th": "2 มิ.ย. 2543",
  "date_of_expiry_en": "1 Jun. 20241 Ju",
  "date_of_expiry_th": "1 มิ.ย. 2567",
  "date_of_issue_en": "18. Apr. 2016",
  "date_of_issue_th": "18 เม.ย. 2559",
  "first_name_en": "Mr. Vitthawat",
  "id_number": "129g 00319 94 6",
  "last_name_en": "Chanphuek",
  "name_th": "นาย วิทวัส จันทร์พฤกษ์",
  "religion_th": "WWE",
  "serial_number": "2399-02-04181012",
}
`;

exports[`ocr 5.jpg return correctly 1`] = `
Object {
  "address_th_1": "599/1 หมู่ที่ 1 ต.วังกระแจะ",
  "address_th_2": ".เมืองตราด จ.ตราด",
  "confidence": Object {
    "address_th_1": 0.94,
    "address_th_2": 0.84,
    "date_of_birth_en": 0.97,
    "date_of_birth_th": 0.97,
    "date_of_expiry_en": 0.74,
    "date_of_expiry_th": 0.73,
    "date_of_issue_en": 0.96,
    "date_of_issue_th": 0.88,
    "first_name_en": 0.98,
    "id_number": 0.92,
    "last_name_en": 0.92,
    "name_th": 0.75,
    "religion_th": 0,
    "serial_number": 0,
  },
  "date_of_birth_en": "19 Nov. 2001",
  "date_of_birth_th": "19 พ.ย. 2544",
  "date_of_expiry_en": "18 Nov. 2020",
  "date_of_expiry_th": "18 W.U. 2563",
  "date_of_issue_en": "24 Dec 2011",
  "date_of_issue_th": "24 2554",
  "first_name_en": "Master Chatpong",
  "id_number": "1 2301 01104 11 8",
  "last_name_en": "Pongcharoen",
  "name_th": "ด.ช. ชัชพงศ์ พงษ์เจริญ",
  "religion_th": "",
  "serial_number": "",
}
`;
